ci/jenkins/test.sh

```diff
@@ -453,13 +456,25 @@ function deliver_antrea_windows {
                 ssh -o StrictHostKeyChecking=no -n Administrator@${IP} "docker load -i antrea-windows.tar.gz"
             fi
         fi
+        # Some tests need us.gcr.io/k8s-artifacts-prod/e2e-test-images/agnhost:2.13 image but it is not for windows/amd64 10.0.17763
```

Comment:
Why move the logic position?

Reply:
Building agent image may lead to deleting some test images since building cache can cause storage limit on some testbeds, this change it to save test images.